File: platforms/android/App/aj/src/main/java/applica/aj/AJ.java
package applica.aj;

import applica.aj.runtime.Plugin;

/**
 * Contains shortcuts to call AJ framework common methods
 */

public class AJ {

    public static Store getStore(String type) {
        return AJApp.runtime().getStore(type);
    }

    public static void subscribe(String store, Object owner, Store.Subscription subscription) {
        AJApp.runtime().subscribe(store, owner, subscription);
    }

    public static void unsubscribe(String store, Object owner) {
        AJApp.runtime().unsubscribe(store, owner);
    }

    public static void registerPlugin(Plugin plugin) {
        AJApp.runtime().registerPlugin(plugin);
    }

    public static Plugin getPlugin(String plugin) {
        return AJApp.runtime().getPlugin(plugin);
    }

    public static void exec(final String plugin, final String fn, final AJObject data, final Plugin.Callback callback) {
        AJApp.runtime().exec(plugin, fn, data, callback);
    }

    public static Semaphore run(final String action) {
        return AJApp.runtime().run(action);
    }

    public static Semaphore run(final String action, final AJObject data) {
        return AJApp.runtime().run(action, data);
    }

}

File: platforms/ios/Libs/AJ/AJ/AJ.swift
//
//  AJ.swift
//  AJ
//
//  Created by Bruno Fortunato on 10/10/16.
//  Copyright © 2016 Bruno Fortunato. All rights reserved.
//

import Foundation

open class AJ {
    
    open class func get(store type: String) -> AJStore {
        return AJApp.runtime().get(store: type)
    }
    
    open class func subscribe(to store: String, owner: AnyObject, subscription: @escaping Subscription) {
        AJApp.runtime().subscribe(to: store, owner: owner, subscription: subscription)
    }
    
    open class func unsubscribe(from store: String, owner: AnyObject) {
        AJApp.runtime().unsubscribe(from: store, owner: owner)
    }
    
    open class func run(action: String, data: AJObject = AJObject.empty()) -> AJSemaphore {
        return AJApp.runtime().run(action: action, data: data)
    }
    
    open class func register(plugin: AJPlugin) {
        AJApp.runtime().register(plugin: plugin)
    }
    
    open class func get(plugin: String) -> AJPlugin {
        return AJApp.runtime().get(plugin: plugin)
    }
    
    open class func exec(plugin: String, fn: String, data: AJObject, callback: AJPluginCallback? = nil) {
        return AJApp.runtime().exec(plugin: plugin, fn: fn, data: data, callback: callback ?? { (s, r) in })
    }
    
}
